features/lorawan/system/lorawan_data_structures.h

```diff
@@ -0,0 +1,2896 @@
+     */
+    uint8_t frame_pending;
+    /** Payload data
+     *
+     * A pointer to the received data stream.
+     */
+    uint8_t buffer[LORAMAC_PHY_MAXPAYLOAD];
+    /** Payload size
+     *
+     * The size of the received data stream.
+     */
+    uint16_t buffer_size;
+    /** RX-data indication
+     *
+     * Indicates, if data is available.
+     */
+    bool rx_data;
+    /** Packet RSSI
+     *
+     * The RSSI of the received packet.
+     */
+    int16_t rssi;
+    /** Packet SNR
+     *
+     * The SNR of the received packet.
+     */
+    uint8_t snr;
+    /** Receive window
+     *
+     * [0: Rx window 1, 1: Rx window 2]
+     */
+    uint8_t rx_slot;
+    /** ACK-received
+     *
+     * Set if an acknowledgement was received.
+     */
+    bool ack_received;
+    /** Downlink counter
+     *
+     * The downlink counter value for the received frame.
+     */
+    uint32_t downlink_counter;
+} lora_mac_mcps_indication_t;
+
+/** lora_mac_rx_message_type_t
+ *
+ * An enum representing a structure for RX messages.
+ */
+typedef enum lora_mac_rx_message_type {
+    LORAMAC_RX_MLME_CONFIRM = 0,    /**< lora_mac_mlme_confirm_t */
+    LORAMAC_RX_MCPS_CONFIRM,        /**< lora_mac_mcps_confirm_t */
+    LORAMAC_RX_MCPS_INDICATION      /**< lora_mac_mcps_indication_t */
+} lora_mac_rx_message_type_t;
+
+/** lora_mac_rx_message_by_type_t union
+ *
+ * A union representing a structure for RX messages.
+ */
+typedef union lora_mac_rx_message_by_type_t {
+    lora_mac_mlme_confirm_t mlme_confirm;
+    lora_mac_mcps_confirm_t mcps_confirm;
+    lora_mac_mcps_indication_t mcps_indication;
+} lora_mac_rx_message_by_type_t;
+
+/** lora_mac_rx_message_t
+ *
+ * A structure representing a structure for an RX message.
+ */
+typedef struct lora_mac_rx_message {
+    bool receive_ready;
+    lora_mac_rx_message_type_t type;
+    lora_mac_rx_message_by_type_t rx_message;
+    uint16_t pending_size;
+    uint16_t prev_read_size;
+} lora_mac_rx_message_t;
+
+/** lora_mac_dr_range_t union
+ *
+ * A union representing a structure for the minimum and maximum data rate.
+ */
+typedef union lora_mac_dr_range {
+    /** Byte-access
+     *
+     * Byte-access to the bits.
+     */
+    int8_t value;
+    /** lora_mac_fields_s
+     *
+     * A structure to store the minimum and maximum data rate.
+     */
+    struct lora_mac_fields {
+        /** Minimum data rate
+         *
+         * EU868 - [DR_0, DR_1, DR_2, DR_3, DR_4, DR_5, DR_6, DR_7]
+         *
+         * US915 - [DR_0, DR_1, DR_2, DR_3, DR_4]
+         */
+        int8_t min :4;
+        /** Maximum data rate
+         *
+         * EU868 - [DR_0, DR_1, DR_2, DR_3, DR_4, DR_5, DR_6, DR_7]
+         *
+         * US915 - [DR_0, DR_1, DR_2, DR_3, DR_4]
+         */
+        int8_t max :4;
+    } lora_mac_fields_s;
+} lora_mac_dr_range_t;
+
+/** LoRaWAN device class definition
+ *
+ */
+typedef enum lora_mac_device_class {
+    LORA_CLASS_A, /**< LoRaWAN device class A */
```

Comment:
Some enum values are confusing, here its `LORA_CLASS_A`, above is `LORAMAC_RX_MLME_CONFIRM ` and below in the events its just `CONNECTED`.

 `LORAMAC_RX_MLME_CONFIRM ` is  correct, to be prefixed to be more descriptive .